code/modules/forging/forge.dm

```diff
@@ -0,0 +1,526 @@
+
+/obj/structure/forge/Initialize()
+	. = ..()
+	START_PROCESSING(SSobj, src)
+
+/obj/structure/forge/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	. = ..()
+
+/**
+ * Here we check both strong (coal) and weak (wood) fuel
+ */
+/obj/structure/forge/proc/check_fuel()
+	if(forge_fuel_strong) //use strong fuel first
+		forge_fuel_strong -= 5 SECONDS
+		target_temperature = 100
+		return
+	if(forge_fuel_weak) //then weak fuel second
+		forge_fuel_weak -= 5 SECONDS
+		target_temperature = 50
+		return
+	target_temperature = 0 //if no fuel, slowly go back down to zero
+
+/**
+ * Here we make the reagent forge reagent imbuing
+ */
+/obj/structure/forge/proc/create_reagent_forge()
+	reagent_forging = TRUE
+	say("gurgle!") //ReplaceWithBalloonAlertLater
+	color = "#ff5151"
+	name = "reagent forge"
+	desc = "[initial(desc)]<br>It has the ability to imbue forged metals with chemicals!"
+
+/**
+ * Here we give a fail message as well as set the in_use to false
+ */
+/obj/structure/forge/proc/fail_message(mob/living/fail_user, message)
+	to_chat(fail_user, "<span class='warning'>[message]</span>")
+	in_use = FALSE
+
+/**
+ * Here we adjust our temp to the target temperature
+ */
+/obj/structure/forge/proc/check_temp()
+	if(forge_temperature > target_temperature) //above temp needs to lower slowly
+		if(sinew_lower_chance && prob(sinew_lower_chance))//chance to not lower the temp, up to 100 from 10 sinew
+			return
+		forge_temperature -= 5
+		return
+	else if(forge_temperature < target_temperature && (forge_fuel_weak || forge_fuel_strong)) //below temp with fuel needs to rise
+		forge_temperature += 5
+
+	if(forge_temperature > 0)
+		icon_state = "forge_full"
+		light_range = 3
+	else if(forge_temperature <= 0)
+		icon_state = "forge_empty"
+		light_range = 0
+
+/**
+ * Here we fix any weird in_use bugs
+ */
+/obj/structure/forge/proc/check_in_use()
+	if(!in_use)
+		return
+	for(var/mob/living/living_mob in range(1,src))
+		if(!living_mob)
+			in_use = FALSE
+
+/**
+ * Here we spawn coal depending on a chance of using wood
+ */
+/obj/structure/forge/proc/spawn_coal()
+	var/obj/item/stack/sheet/mineral/coal/spawn_coal = new(get_turf(src))
+	spawn_coal.name = "charcoal"
+
+/obj/structure/forge/process()
+	if(!COOLDOWN_FINISHED(src, forging_cooldown)) //every 5 seconds to not be too intensive, also balanced around 5 seconds
+		return
+	COOLDOWN_START(src, forging_cooldown, 5 SECONDS)
+	check_fuel()
+	check_temp()
+	check_in_use() //plenty of weird bugs, this should hopefully fix the in_use bugs
+
+/obj/structure/forge/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	var/user_smithing_skill = user.mind.get_skill_level(/datum/skill/smithing)
+	var/previous_level = forge_level
+	if(user_smithing_skill <= SKILL_LEVEL_NOVICE)
+		to_chat(user, "<span class='notice'>[src] requires you to be more experienced!</span>")
+		return
+	if(user_smithing_skill >= SKILL_LEVEL_APPRENTICE)
```

Comment:
what are these even. why does touching the forge fully upgrade its stuff without you having to do anything